custom/enikshay/tasks.py

```diff
@@ -76,31 +91,59 @@ class EpisodeUpdater(object):
     This is applicable to various enikshay domains. The domain can be specified in __init__ method
     """
 
-    def __init__(self, domain):
+    def __init__(self, domain, task_id=None):
         self.domain = domain
+        self.task_id = task_id
         self.updaters = [
             EpisodeAdherenceUpdate,
```

Comment:
I've been thinking the same, actually. I was concerned that splitting it up would then require us to loop through every single case multiple times, instead of just once, but I'm not sure if that is the bottleneck yet.